service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpConnectionHandler.java

```diff
@@ -116,12 +183,14 @@ public void onConnectionBound(Event event)
                 ((TransportInternal)transport).addTransportLayer(webSocket);
             }
 
-            Sasl sasl = transport.sasl();
-            sasl.plain(this.userName, this.sasToken);
```

Comment:
q - so previously we didn't have a dedicated cbs link to send auth tokens over, was this token associated with a session directly?

Reply:
sasl.plain() is moreso a connection level communication, as opposed to a session or link level communication. The old flow was to open the connection, send the username/password, and then we could open the worker session/links